scipy/interpolate/interpolate.py

```diff
@@ -304,14 +304,24 @@ def __call__(self, x, y, dx=0, dy=0, assume_sorted=False):
         return array(z)
 
 
-def _duplicate(ab):
-    # ab is either a pair (a, b) or a single value. In the latter case,
-    # transform to a pair (a, a)
-    try:
-        a, b = ab
-    except TypeError:
-        a, b = ab, ab
-    return a, b
+def _check_broadcast_up_to(shape_from, shape_to, name):
+    """Helper to check that shape_from broadcasts up to shape_to"""
+    if len(shape_to) >= len(shape_from):
+        for t, f in zip(shape_to[::-1], shape_from[::-1]):
+            if f != 1 and f != t:
+                break
+        else:  # all checks pass
+            return
+    # at least one check failed
+    raise ValueError('%s argument must be able to broadcast up '
+                     'to shape %s but had shape %s'
+                     % (name, shape_to, shape_from))
```

Comment:
It could likely work with `broacast_to` (which is new in numpy 1.10 IIRC) in a try-except block?
Anyway, this version seems to work so it's fine IMO.